packages/eslint-plugin/src/rules/no-base-to-string.ts

```diff
@@ -180,6 +180,16 @@ export default createRule<Options, MessageIds>({
           checkExpression(expression);
         }
       },
+      CallExpression(node: TSESTree.CallExpression): void {
+        if (
+          node.callee.type === AST_NODE_TYPES.Identifier &&
+          // eslint-disable-next-line @typescript-eslint/internal/prefer-ast-types-enum
+          node.callee.name === 'String' &&
+          node.arguments[0]
+        ) {
+          checkExpression(node.arguments[0]);
+        }
+      },
```

Comment:
In theory, this could allow any call to a function named `String`... in practice, I don't mind not having extra code to make sure this is the built-in `String`. I think that's a pretty rare & weird edge case.